### PR TITLE
Added Astar SS58 address type

### PR DESCRIPTION
**Pull request by akru**
## Description

Added support for Astar Network
https://astar.network

## Checklist

- [ ] Prefix PR title with `[WIP]` if necessary.
- [ ] Add tests to cover changes as needed.
- [ ] Update documentation as needed.
- [ ] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain
- [ ] If there is a related Issue, mention it in the description (e.g. Fixes #<issue_number> ).

### CODE REVIEW

**Comment by hewigovens** [comment]
Does it really work? please add some real world tx tests

File: registry.json [hunk 1]
```diff
@@ -859,6 +859,30 @@
...
     }
   },
+  {
+    "id": "astar",
+    "name": "Astar",
+    "coinId": 634,
+    "symbol": "ASTR",
+    "decimals": 18,
+    "blockchain": "Polkadot",
+    "derivationPath": "m/44'/434'/0'/0'/0'",
```
Comment (catenocrypt) [inline registry.json:869]: Is 634 the correct coin ID?  In derivation path 434 is used.  Astar is not found in slip-44, and 434 is the Id of Kusama.

**Review by catenocrypt — CHANGES_REQUESTED** [review]

**Comment by FitzLu** [comment]
> Does it really work? please add some real world tx tests

https://astar.subscan.io/extrinsic/0x7d20bd691f5943a98aa05ebada7b45f156edfef53d5a911194d5cb2a1694d979

File: registry.json (hunk 1 above)
Comment (akru) [reply registry.json:869]: Yes, you are right, let me fix it

Comment (akru) [reply registry.json:869]: Waiting for merge https://github.com/satoshilabs/slips/pull/1272

**Comment by hewigovens** [comment]
@FitzLu Please add that tx to test and reuse `Polkadot/Entry`

File: registry.json (hunk 1 above)
Comment (catenocrypt) [reply registry.json:869]: It got merged, with value `810`.  Can you please consolidate the 3 values? (654, 434, 810).  Ideally one should be sued for coin type, in the derivation path, and slip44.

**Comment by akru** [comment]
@hewigovens sample tx added into registry, unfortunately Entry returns custom coin type and can't be reused.

**Comment by hewigovens** [comment]
1. Please check `Cosmos/Entry.h` or `Ethereum/Entry.h` to see how to reuse entry classed
2. I meant swift test case like https://github.com/trustwallet/wallet-core/blob/master/swift/Tests/Blockchains/KusamaTests.swift#L35 and cpp test like https://github.com/trustwallet/wallet-core/blob/master/tests/Polkadot/SignerTests.cpp#L30

**Comment by hewigovens** [comment]
@akru could you please fix the iOS / Android CI? need to update address test `CoinAddressDerivationTests.kt/swift`

**Comment by akru** [comment]
@hewigovens Sure, let me check it

**Review by hewigovens — COMMENTED** [review]
Could you please resolve those review comments?

**Comment by akru** [comment]
@hewigovens Im sorry for delay, I’ll resolve it during this week

**Pull request closed without merging**